packages/kbn-shared-ux-components/src/empty_state/kibana_no_data_page.tsx

```diff
@@ -17,19 +18,27 @@ export interface Props {
 
 export const KibanaNoDataPage = ({ onDataViewCreated, noDataConfig }: Props) => {
   const { hasESData, hasUserDataView } = useData();
+  const [isLoading, setIsLoading] = useState(true);
   const [dataExists, setDataExists] = useState(false);
   const [hasUserDataViews, setHasUserDataViews] = useState(false);
 
   useEffect(() => {
     const checkData = async () => {
       setDataExists(await hasESData());
       setHasUserDataViews(await hasUserDataView());
+      setIsLoading(false);
     };
     // TODO: add error handling
     // https://github.com/elastic/kibana/issues/130913
-    checkData().catch(() => {});
+    checkData().catch(() => {
+      setIsLoading(false);
+    });
   }, [hasESData, hasUserDataView]);
 
+  if (isLoading) {
+    return <EuiLoadingSpinner size="xl" />;
```

Comment:
Can we actually change this to the loading Elastic logo? Since it's loading an entire page view, it's nice to have something larger. Also we should be able to center it with flex.
```suggestion
    return <EuiLoadingElastic css={{ alignSelf: 'center', justifySelf: 'center' }} size="xxl" />;
```

But there's a rogue semicolon lingering on the page somewhere and it's in both Lens and Discover.
![Screen Shot 2022-05-17 at 15 17 34 PM](https://user-images.githubusercontent.com/549577/168896018-1ce9a61c-1140-453c-b40d-9d8b78cd549a.png)
![Screen Shot 2022-05-17 at 15 31 15 PM](https://user-images.githubusercontent.com/549577/168896099-23c4f744-9ae5-4180-88e8-2703dd634e63.png)

Reply:
This is coming from Discover, will fix it there, so I don't bundle it in this PR

Reply:
https://github.com/elastic/kibana/pull/132387

Reply:
Eyy Nice quick fix! Thank you! Since that's merged now, can we merge upstream here so I can check if my centering suggestion works?

Reply:
done!